src/components/Table/common/utils.ts

```diff
@@ -256,35 +236,17 @@ export function getGridTemplateColumns<T extends RowData>(
 /**
  * Returns initial table state.
  * @param columns - Column configuration.
- * @param defaultSort - Column sort configuration.
  * @returns initial table state.
  */
-export function getInitialState(
-  columns: ColumnConfig[],
-  defaultSort: ColumnSort | undefined
+export function getInitialState<T extends RowData>(
+  columns: ColumnConfig<T>[]
 ): InitialTableState {
   const columnVisibility = getInitialTableColumnVisibility(columns);
-  const sorting = getInitialTableStateSorting(defaultSort);
```

Comment:
Removing `sorting` from the function `getInitialState`; we want to manage initial state from table options and as we are dealing with table sorting as part of this feature, I have ensured the initial sorting state can only be defined in table options.